File: atv7/exerc7.go
package main
import "fmt"

type Professor struct {
    Nome string
}

type Escola struct {
    Nome       string
    Professores []Professor
}

func (e *Escola) adicionarProfessor(p Professor) {
    e.Professores = append(e.Professores, p)
}

func (e Escola) listarProfessores() {
    fmt.Println("Professores da escola", e.Nome + ":")
    for _, p := range e.Professores {
        fmt.Println(p.Nome)
    }
}

func main() {
    escola := Escola{Nome: "UNIPÊ"}
    p1 := Professor{Nome: "Professor João"}
    p2 := Professor{Nome: "Professor Maria"}

    escola.adicionarProfessor(p1)
    escola.adicionarProfessor(p2)

    escola.listarProfessores()
}

File: atv7/exerc7.py
class Professor:
    def __init__(self, nome):
        self.nome = nome

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)

    def listar_professores(self):
        print(f"Professores da escola {self.nome}:")
        for professor in self.professores:
            print(professor.nome)

escola = Escola("UNIPÊ")
p1 = Professor("Professor João")
p2 = Professor("Professor Maria")

escola.adicionar_professor(p1)
escola.adicionar_professor(p2)

escola.listar_professores()
